template/src/pageComponents/mint/ContractDemo.tsx

```diff
@@ -48,14 +72,16 @@ export default function MintContractDemo() {
     return <SwitchNetwork />;
   }
 
-  if (isLoading) {
-    // A future enhancement would be a nicer spinner here.
-    return <span className="text-xl">loading...</span>;
+  if (isLoadingCollectionMetadata || contract.status !== 'ready') {
+    return (
+      <div className="my-5 flex justify-center align-middle">
+        <span className="text-xl">
+          <SpinnerIcon className="h-20 w-20 animate-spin" />
+        </span>
```

Comment:
I removed the comment, I think this is good as is :)

Reply:
Thanks. I do think a wrapper can be added, something like <Loader /> where tailwind's animation classes are already added so we don't need to add it every time. But maybe this can be done in a separate pr

Reply:
yea, I was thinking that too but yea maybe we can do this in a separate PR. 

CC @arsood @Zizzamia  for later